go-minitwit/api_handlers.go

```diff
@@ -397,24 +397,24 @@ func apiFllwsHandler(c *gin.Context) {
 			// Convert requestBody.Unfollow to profileUserID
 			profileUserID, err := getUserIDByUsername(requestBody.Unfollow)
 			if err != nil || profileUserID == -1 {
-				c.AbortWithStatus(http.StatusBadRequest)
+				c.AbortWithStatus(http.StatusNotFound)
 				return
 			}
 			profileUserIDStr := strconv.FormatInt(profileUserID, 10)
 
 			// Unfollow the user
 			if err := unfollowUser(userIdStr, profileUserIDStr); err != nil {
-				errorData.status = http.StatusInternalServerError
+				errorData.status = http.StatusNotFound
```

Comment:
I would also change this error since both users were found previously. The same thing can be applied to the Follow process.